src/tests/functional/shared_test_classes/src/single_op/roi_align.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/single_op/roi_align.hpp"
+
+#include "openvino/core/enum_names.hpp"
+
+namespace ov {
+namespace test {
+std::string ROIAlignLayerTest::getTestCaseName(const testing::TestParamInfo<roialignParams>& obj) {
+    std::vector<InputShape> input_shapes;
+    ov::Shape coords_shape;
+    int pooled_h;
+    int pooled_w;
+    float spatial_scale;
+    int pooling_ratio;
+    std::string pooling_mode;
+    ov::element::Type model_type;
+    std::string target_device;
+    std::tie(input_shapes, coords_shape, pooled_h, pooled_w, spatial_scale,
+    pooling_ratio, pooling_mode, model_type, target_device) = obj.param;
+
+    std::ostringstream result;
+    result << "IS=(";
+    for (size_t i = 0lu; i < input_shapes.size(); i++) {
+        result << ov::test::utils::partialShape2str({input_shapes[i].first})
+               << (i < input_shapes.size() - 1lu ? "_" : "");
+    }
+    result << ")_TS=";
+    for (size_t i = 0lu; i < input_shapes.front().second.size(); i++) {
+        result << "{";
+        for (size_t j = 0lu; j < input_shapes.size(); j++) {
+            result << ov::test::utils::vec2str(input_shapes[j].second[i]) << (j < input_shapes.size() - 1lu ? "_" : "");
+        }
+        result << "}_";
+    }
+    result << "coordShape=" << ov::test::utils::vec2str(coords_shape) << "_";
+    result << "pooledH=" << pooled_h << "_";
+    result << "pooledW=" << pooled_w << "_";
+    result << "spatialScale=" << spatial_scale << "_";
+    result << "poolingRatio=" << pooling_ratio << "_";
+    result << "poolingMode=" << pooling_mode << "_";
+    result << "modelType=" << model_type.to_string() << "_";
+    result << "trgDev=" << target_device;
+    return result.str();
+}
+
+static int randInt(int low, int high) {
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<int> dis(low, high);
+    return dis(gen);
+}
+
+void ROIAlignLayerTest::fillCoordTensor(std::vector<float>& coords, int height, int width,
```

Comment:
Hm. Coordinates? In this case why we use `float`? Looks like should be renamed

Reply:
I have no idea how it should be renamed, so will port as is. If you have some suggestions, I'm open to apply it